File: ProblemSet/366. 斐波纳契数列/366. 斐波纳契数列 公式法.java
public class Solution {
    /**
     * @param n: an integer
     * @return: an ineger f(n)
     */
    public int fibonacci(int n) {
        double glodenRatio = (1 + Math.sqrt(5)) / 2;
        return (int)Math.round(Math.pow(glodenRatio, n - 1) / Math.sqrt(5));
    }
}

File: ProblemSet/366. 斐波纳契数列/python/366. 斐波纳契数列 公式法.py
class Solution:
    """
    @param n: an integer
    @return: an ineger f(n)
    """
    def fibonacci(self, n):
        gloden_ratio = (1 + 5 ** 0.5) / 2
        return round(gloden_ratio ** (n - 1)  / (5 ** 0.5))
